src/lj_alloc.c

```diff
@@ -379,7 +379,7 @@ static void *CALL_MREMAP_(void *ptr, size_t osz, size_t nsz, int flags)
 #define CALL_MREMAP(addr, osz, nsz, mv) CALL_MREMAP_((addr), (osz), (nsz), (mv))
 #define CALL_MREMAP_NOMOVE	0
 #define CALL_MREMAP_MAYMOVE	1
-#if LJ_64 && !LJ_GC64
+#if LJ_TARGET_ARM64 || (LJ_64 && !LJ_GC64)
```

Comment:
This may have far-reaching impact on the whole JIT compiler on ARM64. It'll need special care.